dlpy/applications/efficientnet.py

```diff
@@ -0,0 +1,292 @@
+            x = GlobalAveragePooling2D(name=prefix + "global_avg_pool")(x)
+            # Squeeze
+            channels_se = max(1, int(in_channels * se_ratio))
+            x = Conv2d(channels_se, 1, include_bias=True, act=activation_fn, name=prefix + 'squeeze')(x)
+            x = Conv2d(n_groups, 1, include_bias=True, act='sigmoid', name=prefix + 'excitation')(x)
+            x = Reshape(name=prefix + 'reshape', width=n_groups, height=1, depth=1)(x)
+            x = Scale(name=prefix + 'scale')([se_input, x])  # x = out*w
+
+        # Project
+        x = Conv2d(out_channels, 1, include_bias=False, act='identity', name=prefix + 'project')(x)
+        x = BN(name=prefix + 'project_BN', act='identity')(x)  # identity activation on narrow tensor
+        # Prepare output for MBConv block
+        if in_channels == out_channels and stride == 1 and (not noskip):
+            # dropout can be added.
+            return Res(name=prefix + 'add_se_residual')([x, inputs])
+        else:
+            return x
+
+    parameters = locals()
+    input_parameters = get_layer_options(input_layer_options, parameters)
+    inp = Input(**input_parameters, name='data')
+    # refer to Table 1  "EfficientNet-B0 baseline network" in paper:
+    # "EfficientNet: Rethinking Model Scaling for Convolutional Neural Networks"
+    stage_id = 0
+    out_channels = round_filters(32, width_coefficient,
+                                 depth_divisor)  # multiply with width multiplier: width_coefficient
+    x = Conv2d(out_channels, 3, stride=2, include_bias=False, name='Conv1', act='identity')(inp)
+    x = BN(name='bn_Conv1', act=activation_fn)(x)
+
+    # Create stages with MBConv blocks from stage 1
+    in_channels = out_channels  # number of input channels for first MBblock
+    stage_id +=1
+    total_blocks = float(sum(args[2] for args in blocks_args))
+    for expansion, out_channels, num_blocks, ksize, stride, se_ratio in blocks_args:
+        out_channels = round_filters(out_channels, width_coefficient, depth_divisor)
+        num_blocks = round_repeats(num_blocks, depth_coefficient)
+        strides = [stride] + [1] * (num_blocks - 1)
+        for block_id, stride in enumerate(strides):
+            x = _MBConvBlock(x, in_channels, out_channels, ksize, stride, expansion, se_ratio, stage_id, block_id,activation_fn)
+            in_channels = out_channels  # out_channel
+        stage_id += 1
+
+    last_block_filters = round_filters(1280, width_coefficient, depth_divisor)
+    x = Conv2d(last_block_filters, 1, include_bias=False, name='Conv_top', act='identity')(x)
+    x = BN(name='Conv_top_bn', act=activation_fn)(x)
+
+    x = GlobalAveragePooling2D(name="Global_avg_pool", dropout=dropout_rate)(x)
+    x = OutputLayer(n=n_classes)(x)
+
+    model = Model(conn, inp, x, model_table)
+    model.compile()
+    return model
+
+
+def EfficientNetB0(conn, model_table='EfficientNetB0',
+                   n_classes=1000,
+                   **kwargs):
```

Comment:
why do you need kwargs?

Reply:
The kwargs can be passed to the function call EfficientNet() within EfficientNetB0. 
The following example shows that the parameter "random_crop" in function EfficentNetB0 is used and passed to function EfficientNet:
effnetb0 = EfficientNetB0(s, model_table='EfficientNetB0', n_classes=10, random_crop='resizethencrop', random_flip='h')